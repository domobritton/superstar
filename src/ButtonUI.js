import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import StarRate from '@material-ui/icons/StarRate';

import styled from 'styled-components';

// will need logic for button that will star and unstar repos

export default class ButtonUI extends Component {
  state = {
    clicked: false,
  };

  handleClick = () => {
    this.setState(prevState => ({
      clicked: !prevState.clicked,
    }));
  };
  render() {
    const { clicked } = this.state;
    const starColor = clicked ? { color: 'gold' } : { color: 'white' };
    return (
      <Button
        onClick={this.handleClick}
        variant="contained"
        color="primary"
        style={{ width: 120 }}
      >
        <Star style={starColor} />
        {clicked ? 'STAR' : 'UNSTAR'}
      </Button>
    );
  }
}

const Star = styled(StarRate)`
  margin-right: auto;
`;
