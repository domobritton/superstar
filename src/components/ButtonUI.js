import React, { Component } from 'react';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import { withApollo, Mutation } from 'react-apollo';

import Button from '@material-ui/core/Button';
import StarRate from '@material-ui/icons/StarRate';
import styled from 'styled-components';

const ADD_STAR = gql`
  mutation AddStar($input: AddStarInput!) {
    addStar(input: $input) {
      starrable {
        id
        viewerHasStarred
      }
    }
  }
`;

const REMOVE_STAR = gql`
  mutation RemoveStar($input: RemoveStarInput!) {
    removeStar(input: $input) {
      starrable {
        id
        viewerHasStarred
      }
    }
  }
`;

class ButtonUI extends Component {
  render() {
    const { id, starred } = this.props;
    const starColor = starred ? { color: 'gold' } : { color: 'white' };
    return (
      <div>
        {starred ? (
          <Mutation mutation={REMOVE_STAR} variable={{ input: id }}>
            {(removeStar, { data }) => (
              <Button
                onClick={e => {
                  e.preventDefault();
                  removeStar({
                    variables: { input: { starrableId: this.props.id } },
                  });
                }}
                variant="contained"
                color="primary"
                style={{ width: 120 }}
              >
                <Star style={starColor} />
                {console.log(data)}
                UNSTAR
              </Button>
            )}
          </Mutation>
        ) : (
          <Mutation mutation={ADD_STAR} variable={{ input: id }}>
            {(addStar, { data }) => (
              <Button
                onClick={e => {
                  e.preventDefault();
                  addStar({
                    variables: { input: { starrableId: this.props.id } },
                  });
                }}
                variant="contained"
                color="primary"
                style={{ width: 120 }}
              >
                <Star style={starColor} />
                {console.log(data)}
                STAR
              </Button>
            )}
          </Mutation>
        )}
      </div>
    );
  }
}

ButtonUI.propTypes = {
  Id: PropTypes.number,
  starred: PropTypes.bool,
};

const Star = styled(StarRate)`
  margin-right: auto;
`;

export default withApollo(ButtonUI);
