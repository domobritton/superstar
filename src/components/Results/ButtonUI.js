import React from 'react';
import PropTypes from 'prop-types';
import { withApollo, Mutation } from 'react-apollo';

import { ADD_STAR, REMOVE_STAR } from '../../utilities/graphql/star';

import Button from '@material-ui/core/Button';
import { Star } from './ButtonUIStyle';

const ButtonUI = ({ id, starred }) => {
  const starColor = starred ? { color: 'gold' } : { color: 'white' };
  return (
    <div>
      {starred ? (
        <Mutation mutation={REMOVE_STAR} variable={{ input: id }}>
          {(removeStar, { data }) => (
            <Button
              data-testid={'mutate'}
              onClick={e => {
                e.preventDefault();
                removeStar({
                  variables: { input: { starrableId: id } },
                });
              }}
              variant="contained"
              color="primary"
              style={{ width: 120 }}
            >
              <Star style={starColor} />
              UNSTAR
            </Button>
          )}
        </Mutation>
      ) : (
        <Mutation mutation={ADD_STAR} variable={{ input: id }}>
          {(addStar, { data }) => (
            <Button
              data-testid={'mutate'}
              onClick={e => {
                e.preventDefault();
                addStar({
                  variables: { input: { starrableId: id } },
                });
              }}
              variant="contained"
              color="primary"
              style={{ width: 120 }}
            >
              <Star style={starColor} />
              STAR
            </Button>
          )}
        </Mutation>
      )}
    </div>
  );
};

ButtonUI.propTypes = {
  Id: PropTypes.number,
  starred: PropTypes.bool,
};

export default withApollo(ButtonUI);
