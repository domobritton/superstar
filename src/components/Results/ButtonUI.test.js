import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { ApolloProvider } from 'react-apollo';
import ButtonUI from './ButtonUI';

afterEach(cleanup);

const clientMock = jest.fn();

test('<ButtonUI />', () => {
  const { getByTestId, container } = render(
    <ApolloProvider client={{}}>
      <ButtonUI />
    </ApolloProvider>
  );

  const starButton = getByTestId('mutate');

  expect(container.firstChild).toMatchSnapshot();
  expect(starButton.textContent).toBe('STAR');
});
