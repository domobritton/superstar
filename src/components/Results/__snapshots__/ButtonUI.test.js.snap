// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonUI /> 1`] = `
<div>
  <button
    class="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
    data-testid="mutate"
    style="width: 120px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-30 sc-bdVaJa kjEbUR"
        focusable="false"
        role="presentation"
        style="color: white;"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 11.3l3.71 2.7-1.42-4.36L15 7h-4.55L9 2.5 7.55 7H3l3.71 2.64L5.29 14z"
          transform="scale(1.33, 1.33)"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
          transform="scale(1.33, 1.33)"
        />
      </svg>
      STAR
    </span>
    <span
      class="MuiTouchRipple-root-39"
    />
  </button>
</div>
`;
